src/main/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/controller/BalanceController.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.hmpps.visitallocationapi.controller
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.visitallocationapi.dto.PrisonerBalanceDto
+import uk.gov.justice.digital.hmpps.visitallocationapi.service.BalanceService
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+
+const val VO_BALANCE = "/visits/allocation/prisoner/{prisonerId}/balance"
+
+@RestController
+class BalanceController(val balanceService: BalanceService) {
+  @PreAuthorize("hasRole('ROLE_VISIT_ALLOCATION_API__PRISONER_BALANCE')")
+  @GetMapping(VO_BALANCE)
+  @Operation(
+    summary = "Endpoint to get a prisoners current balance. Returns 0 if prisoner not found",
+    description = "Takes a prisoner id and return their current visit order balance.",
+    responses = [
+      ApiResponse(
```

Comment:
would it be better to return a 204 if prisoner not found? Will be easier to distinguish between prisoner not found and 0. This is how prison API have done it - https://github.com/ministryofjustice/prison-api/blob/3702b4319b14f09ac2c691fa3fa9eba346bc8d63/src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java#L765